agent/config/default.go

```diff
@@ -98,6 +98,7 @@ func DefaultSource() Source {
 			http_max_conns_per_client = 200
 			https_handshake_timeout = "5s"
 			rpc_handshake_timeout = "5s"
+			rpc_read_timeout = "10m"
```

Comment:
10 minutes is really long. Is there a reason why we chose this?

Reply:
I do agree it is long but wanted to set a high upper bound so this config largely remains out of people's concerns. The only data point I have is from a customer who reported `25s` query times. Maybe `1m` is more pragmatic?

Reply:
Yeah, I'm not sure. If a customer was reaching a 25s query, then I suppose we should be careful and try to keep it at a larger number. So 10m is probably appropriately long to not affect anyone.

Everything is being served from in-memory, so I'm not sure if it really should ever be taking that long though. It would be nice if we had good stats around this -- we could set what we believe to be a sane default, rather than the upper bound.

Reply:
Good thoughts here folks. I think 60s seems like a conservative enough default - if your RPCs are taking more than that, the server is in a pretty awful shape and clients should probably be timing out already.

We could consider documenting in our production guide that this should be reduced ideally to just a few seconds.

It would be ideal if this config is reloadable too so that it's easy if it is set too short and causing amplification due to bad retry behaviour or something that the operator can get out of jail without rolling all servers! You probably did that already @kisunji -- i've not looked yet but will leave the note here to come back to!

Reply:
I haven't made it reloadable but it would definitely be a nice UX for operators! Will add a commit